mne/stats/regression.py

```diff
@@ -392,3 +396,36 @@ def _make_evokeds(coefs, conds, cond_length, tmin_s, tmax_s, info):
             kind='average')  # nave and kind are technically incorrect
         cumul += tmax_ - tmin_
     return evokeds
+
+
+def _solver(X, y):
+    """Return b for y=XB for a sparse X and multiple y using a cholesky
+    factorization of the normal equations.
+    Can be used as a solver for ``linear_regression_raw``."""
+    a = (X.T * X).toarray()  # dot product of sparse matrices
+    return linalg.solve(a, X.T * y.T, sym_pos=True,
+                        overwrite_a=True, overwrite_b=True).T
+
+
+def _incremental_solver(X, y):
+    """Return b for y=XB for a sparse X and multiple y using an 
+    incrementally constructed cholesky factorization of the normal
+    equations. This allows to fit nearly arbitrary large models. The
+    general approach is due to Nathaniel J. Smith.
+    Can be used as a solver for ``linear_regression_raw``."""
+    stride_ratio = 2
+    n_times, n_coefs = X.shape
+    one_stride = int(n_coefs * stride_ratio)
+    loops = (n_times // one_stride) - 1
+
+    XtX_all = np.zeros((n_coefs, n_coefs))
+    XtY_all = np.zeros((n_coefs, y.shape[0]))
+    for ii in range(0, n_times - one_stride + 1, one_stride):
+        reach = (one_stride + (n_times % one_stride)
+                 if ii + (one_stride * stride_ratio) > n_times
+                 else one_stride)
+        XtX_all += (X[ii:ii + reach, :].T * X[ii:ii + reach, :])
+        XtY_all += (X[ii:ii + reach, :].T * y[:, ii:ii + reach].T)
```

Comment:
It's approx. 4x slower - the main thing is memory. The matrices can be huge otherwise and crash people's computers.